packages/core/src/useNavigationHelpers.tsx

```diff
@@ -82,4 +94,27 @@ export default function useNavigationHelpers<
     setState,
     onAction,
   ]);
+
+  const { addNavigationListener } = React.useContext(NavigationFocusedContext);
+  const listener = React.useRef((callback: FocusedNavigationCallback) => {
```

Comment:
Why `useRef` here? Maybe you wanted to use `useCallback`? Using ref is incorrect since the values will never update.